ethereum-network-partition-test/ethereum_helpers.star

```diff
@@ -5,9 +5,13 @@ LATEST_BLOCK_NUMBER_GENERIC = "latest"
 # and pad the hexadecimal string to 20 characters (which should be a limit we hopefully never hit)
 # This is a hack to get the hexadecimal block numbers to be comparable between each other
 # We have the equivalent function in Starlark below (see `pad`)
-HEX_PAD_NUMBER = 20
-HEX_PAD_NUMBER_FOR_JQ = HEX_PAD_NUMBER + 2  # +2 to account for the additional `0x` at the beginning of the hex string
-JQ_PAD_HEX_FILTER = """def pad(i): (({0} - (i | length)) * "0") + (i | sub("0x"; "")); pad({1})"""
+JQ_PAD_HEX_FILTER = """
+{} |
+ascii_upcase 
+split("") |
+map({{"0": 0, "1": 1, "2": 2, "3": 3, "4": 4, "5": 5, "6": 6, "7": 7, "8": 8, "9": 9, "A": 10, "B": 11, "C": 12, "D": 13, "E": 14, "F": 15}}[.]) |
```

Comment:
This is a hell of a hack -.-. Not sure whether i feel better about this one or the above pad() one :D 

nit: Could you maybe enclose this logic in a function with an explicit name about what it's doing, like the `pad` func in the previous filter. Will probably help understand this thing in the future.

Reply:
Done! I believe this is better because we can use the correct type in Starlark, allowing us to `<`, `>`, `=`